spec/support/vault_server.rb

```diff
@@ -37,39 +41,45 @@ def initialize
         io.close
         io.unlink
       end
+      wait_for_ready
+      puts "vault server is ready"
+      # sleep to get unseal token
+      sleep 5
 
-      wait_for_ready do
-        @token = File.read(TOKEN_PATH)
+      @token = "root"
 
-        output = ""
-        while
-          io.rewind
-          output = io.read
-          break if !output.empty?
-        end
+      output = ""
+      while io.rewind
+        output = io.read
+        break unless output.empty?
+      end
 
-        if output.match(/Unseal Key.*: (.+)/)
-          @unseal_token = $1.strip
-        else
-          raise "Vault did not return an unseal token!"
-        end
+      if output.match(/Unseal Key.*: (.+)/)
+        @unseal_token = $1.strip
+      else
+        raise "Vault did not return an unseal token!"
       end
     end
 
     def address
       "http://127.0.0.1:8200"
     end
 
-    def wait_for_ready(&block)
-      Timeout.timeout(15) do
-        while !File.exist?(TOKEN_PATH)
-          sleep(0.25)
+    def wait_for_ready
+      uri = URI(address + "/v1/sys/health")
+      Timeout.timeout(10) do
+        loop do
+          begin
+            response = Net::HTTP.get_response(uri)
+            return true if response.code != 200
```

Comment:
I don't really know ruby syntax but would this return false on non-200 response code or not return at all?

Reply:
would not return at all 
perhaps 
```suggestion
            if response.code != 200
              return true
            end
```

would be better?

Reply:
Yes, definitely more clear for a ruby novice like me 😄